### PR TITLE
fixed an issue with min api 24 apps using torque

### PR DESCRIPTION
**Pull request by KennethNickles**
Found an issue with parsing dex files when migrating to min api 24. Updated the dex parsing library which caused some api changes. Also updated some arg names to be a bit more descriptive of what they are actually doing.

### CODE REVIEW

File: torque-core/src/test/kotlin/com/workday/torque/ModuleTestParserSpec.kt [hunk 1]
```diff
@@ -13,241 +14,259 @@ import org.jetbrains.spek.api.dsl.given
...
-        }
-        val targetApkPath = "some_app_path"
-        val testApkPath = "some_path"
-        val testModuleInfo = TestModuleInfo(ModuleInfo(testPackage, testApkPath), testRunner, ModuleInfo(targetPackage, targetApkPath))
-
-        given("no Annotations and with default Ignore NotAnnotations") {
-            val args = Args().apply {
-                testApkPaths = listOf(testApkPath)
-                appApkPath = targetApkPath
+        {
```
Comment (hujim) [inline torque-core/src/test/kotlin/com/workday/torque/ModuleTestParserSpec.kt:17]: might have avoided the huge diff if this indention is moved back
I know, Spek is dumb here lol

File: torque-core/src/main/kotlin/com/workday/torque/Args.kt [hunk 2]
```diff
@@ -25,20 +25,20 @@ data class Args(
...
 
         @Parameter(
-                names = ["--annotations"],
+                names = ["--allowed-annotations"],
                 variableArity = true,
                 description = "Run only test methods WITH specified annotations. If multiple annotations are specified, will only run tests that are annotated with all of them. " +
                         "None by default and runs all tests. If used with other options, will run the intersection of them."
         )
-        var annotations: List<String> = emptyList(),
+        var allowedAnnotations: List<String> = emptyList(),
```
Comment (hujim) [inline torque-core/src/main/kotlin/com/workday/torque/Args.kt:33]: just a thought, how about included/excluded?

**Review by hujim — APPROVED** [review]

File: torque-core/src/main/kotlin/com/workday/torque/Args.kt (hunk 2 above)
Comment (KennethNickles) [reply torque-core/src/main/kotlin/com/workday/torque/Args.kt:33]: yeah that's better, I'll update

File: torque-core/src/test/kotlin/com/workday/torque/ModuleTestParserSpec.kt (hunk 1 above)
Comment (KennethNickles) [reply torque-core/src/test/kotlin/com/workday/torque/ModuleTestParserSpec.kt:17]: My bad, I'll reformat this

File: torque-core/src/test/kotlin/com/workday/torque/utils/TestUtils.kt [hunk 3]
```diff
@@ -1,7 +1,55 @@
...
+    assertEquals(inherited, other.inherited)
+    values.forEach { (key, value) ->
+        assertTrue(other.values.containsKey(key))
+        val otherValue = other.values.getValue(key)
+        value.assertDecodeValue(otherValue)
+    }
+}
+
+private fun DecodedValue.assertDecodeValue(other: DecodedValue) {
+    if(this is DecodedValue.DecodedArrayValue && other is DecodedValue.DecodedArrayValue) {
```
Comment (KennethNickles) [inline torque-core/src/test/kotlin/com/workday/torque/utils/TestUtils.kt:50]: This is actually a bug in the dex parser library, I might open a pr for them. But they didn't override equals for the array comparison so all assert equals fail on DecodedValue types with arrays.

**Pull request merged**